Make chatwoot template easier to use for end users

kfirmanty opened the pull request:
Variables are now exposed in stack so it should be easy for users to deploy their own stack just by overriding few selected variables (included example in README.md)

Tested SMTP with mailtrap and it worked.

kfirmanty committed:
fix: fixed websockets in chatwoot template

kfirmanty committed:
used monk hub postgres and redis template

kfirmanty committed:
fixed depends section

kfirmanty committed:
removed unused templates

kfirmanty committed:
moved variables to process group

kfirmanty committed:
redis entrypoint will now use password from variable

kfirmanty committed:
fix variables name case

kfirmanty committed:
updated README.md with example of overriding variables

kfirmanty committed:
removed redundant metadata, reintroduced nginx to stack, closed port 3000

kfirmanty committed:
added overriding of server-name (used by nginx) variable in README.md example

kfirmanty committed:
moved common section with metadata to top, fixed ssl

kfirmanty committed:
fix smtp authentication - tested with mailtrap

kfirmanty committed:
extended README.md example with variables that were used for working deployment

kfirmanty committed:
added 2 env variables to make it work with sendgrid

kfirmanty commented:
updated template with `smtp-enable-startttls-auto` and `smtp-domain` variables. Tested template with `sendgrid` credentials and successfully managed to send email from chatwoot panel.

sojan-official commented on chatwoot.yaml:
```diff
@@ -11,51 +11,133 @@ common:
     publisher: "monk.io"
     icon: https://storage.googleapis.com/template_logos/chatwoot-image.png
-
+    email: maciej@monk.io
```
we can update the email. 

```suggestion
    email: hello@chatwoot.com
```

sojan-official commented on chatwoot.yaml:
```diff
@@ -11,51 +11,133 @@ common:
+        - <- `SMTP_ENABLE_STARTTLS_AUTO=${smtp-enable-startttls-auto}`
+        - <- `SMTP_DOMAIN=${smtp-domain}`
+        - <- `REDIS_PASSWORD=${redis-password}`
+        - <- `FRONTEND_URL=${server-name}`
```
The expectation of FRONTEND_URL is to have the full URL include `https://` .  I think  server-name returns just the domain name right ? .  I am not sure whether this will break anything. 

should we update ?

kfirmanty replied:
good catch, I will update it. I will setup test domain in cloudflare today and will test it against it.

sojan-official commented on chatwoot.yaml:
```diff
@@ -117,34 +198,66 @@ sidekiq:
+    redis-password:
+      env: REDIS_PASSWORD
+      value: "CHANGE_ME"
+    sidekiq-auth-username:
```
sidekiq-auth-username && sidekiq-auth-password variables are deprecated now. 
so they aren't required

sojan-official reviewed (approved):
added a couple of comments. 

otherwise LGTM.

kfirmanty committed:
removed deprecated sidekiq variables, updated email in metadata

kfirmanty committed:
exposed frontend-url as separate variable

kfirmanty committed:
added dependency on sidekiq for chatwoot

kfirmanty committed:
updated README.md with new variables

kfirmanty commented:
Hi,
So I removed sidekiq variables, exposed frontend-url and added dependency on sidekiq for chatwoot definition so it will always start in proper order. I have tested it with custom domain and both real time messaging and email (sendgrid and mailtrap) worked properly.
If there is any other service integration that you would like me to test to make sure it works 100% correctly please let me know, but in general the template should be ready to go.
There is an example in README.md how end users could deploy their own chatwoot stack but if anything is still not clear let me know and I will try to improve it so they will have great experience using it.
